JavaScript/AppModulesConventions.md

```diff
@@ -0,0 +1,105 @@
+# Wikia App JavaScript modules conventions
+
+This styleguide defines the JavaScript modules strucutre conventions for Wikia App (https://github.com/Wikia/app).
+
+## TOC
+
+* [Files/directories structure](#filesdirectories-structure)
+* [Entry point files](#entry-point-files)
+* [Modules](#modules)
+* [Tracking](#tracking)
+
+
+## Files/directories structure
+
+### Keep all JS in scripts folder
+
+Keep all JavaScript files in `scripts` folder which should be located in extension's root directory (`extensions/wikia/MyExtensions/scripts`)
+
+### Entry-point files
+
+All entry points file-names should start with `index.`. If there is only one entry point file it should be named `index.js`
+
+```javascript
+// bad
+foo.js
+entry.js
+start.js
+moduleName.js
+
+// good
+index.js
+
+// good only if there are more than one entry point files
+index.example.js
+index.foo.js
+```
+
+## Entry point files
+
+### Keep them small
+
+Keep entry point files as small as possible. They should usually just execute some initialization functions from other files.
+```
+TODO: what if we want to have just one js file in extension?
```

Comment:
I think there is no need to create multiple files if we want to for example add just one button click handler.